solidity/app/contracts/Router.sol

```diff
@@ -148,4 +148,11 @@ abstract contract Router is AbacusConnectionClient, IMessageRecipient {
             _interchainGasPaymaster().payGasFor{value: _gasPayment}(leafIndex);
         }
     }
+
+    /**
+     * @notice Calls checkpoint on the outbox
+     */
+    function _checkpoint_on_outbox() internal {
```

Comment:
nit: `_checkpointOnOutbox`, you've been writing too much Rust 😛 

Some other thoughts:
1. Why not just `_checkpoint`?
2. I do think it's worth having some helper functions that dispatch and checkpoint. My motivation is both for developer experience (I think it's important to avoid a user having to write more than one line of code just to send a message) and for gas reasons, because separating the `dispatch` and `checkpoint` calls to the Outbox increase gas costs. Each `_outbox()` call from Router.sol will SLOAD the `abacusConnectionManager` and call `outbox()` on the connection manager, which itself then SLOADs the `outbox` address. Naming-wise in here, I've never been a big fan of `_dispatchToRemoteRouter`, I wonder if we could make things less of a mouthful if we just change that to `_dispatch`, change `_dispatchToRemoteRouterWithGas` to `_dispatchWithGas`, and then we do something like `_dispatchWithGasAndCheckpoint`. Still not amazing but a little less of an eyesore
3. I wonder if we should also provide a `dispatchAndCheckpoint` function in `Outbox.sol`-- it sounds like we expect dapps to always checkpoint, and the vast majority imo will dispatch only a single message, so I think there's an argument to allow dapps to make only 1 call to the Outbox instead of two
4. I like the spirit of `send` but I worry introducing new jargon on top of dispatch / checkpoint may be confusing for developers